File: extract-colors.js
const fs = require('fs');
const path = require('path');
const { createCanvas, loadImage } = require('canvas');

async function extractColor(imagePath) {
    const img = await loadImage(imagePath);
    const canvas = createCanvas(img.width, img.height);
    const ctx = canvas.getContext('2d');

    ctx.drawImage(img, 0, 0);

    // Get pixel data from center of image
    const centerX = Math.floor(img.width / 2);
    const centerY = Math.floor(img.height / 2);
    const imageData = ctx.getImageData(centerX, centerY, 1, 1);
    const [r, g, b] = imageData.data;

    // Convert to hex
    const hex = '#' + [r, g, b].map(x => x.toString(16).padStart(2, '0')).join('');

    return hex;
}

async function extractAllColors() {
    const colorsDir = path.join(__dirname, 'colors');
    const files = fs.readdirSync(colorsDir)
        .filter(f => f.match(/^B\d{2}\.png$/))
        .sort();

    const colors = [];

    for (const file of files) {
        const filePath = path.join(colorsDir, file);
        const hex = await extractColor(filePath);
        const name = file.replace('.png', '');

        colors.push({
            name: name,
            hex: hex,
            image: `/colors/${file}`
        });

        console.log(`${name}: ${hex}`);
    }

    console.log('\n\nColors array for App.jsx:');
    console.log(JSON.stringify(colors, null, 2));
}

extractAllColors().catch(console.error);

File: extract-colors.py
from PIL import Image
import os
import json

def extract_color(image_path):
    """Extract the dominant color from a PNG file"""
    img = Image.open(image_path)
    # Convert to RGB if necessary
    if img.mode != 'RGB':
        img = img.convert('RGB')
    
    # Get pixel from center of image
    width, height = img.size
    center_x, center_y = width // 2, height // 2
    r, g, b = img.getpixel((center_x, center_y))
    
    # Convert to hex
    hex_color = f'#{r:02x}{g:02x}{b:02x}'
    return hex_color

def main():
    colors_dir = 'colors'
    files = sorted([f for f in os.listdir(colors_dir) if f.startswith('B') and f.endswith('.png')])
    
    colors = []
    
    print("Extracted Colors:")
    print("-" * 40)
    
    for file in files:
        filepath = os.path.join(colors_dir, file)
        hex_color = extract_color(filepath)
        name = file.replace('.png', '')
        
        color_obj = {
            'name': name,
            'hex': hex_color,
            'image': f'/colors/{file}'
        }
        
        colors.append(color_obj)
        print(f"{name}: {hex_color}")
    
    print("\n\nColors array for App.jsx:")
    print(json.dumps(colors, indent=2))

if __name__ == '__main__':
    main()
